### PR TITLE
Add support for dry run

### PR DESCRIPTION
**Pull request by ToddBradley**
This PR adds support for a command line option `--dry-run` to do everything except download content.

### CODE REVIEW

File: drpg/sync.py [hunk 1]
```diff
@@ -71,25 +72,30 @@ def sync(self) -> None:
...
+            file_response = httpx.get(
+                url_data["download_url"], timeout=30.0, follow_redirects=True
             )
-            return
-
-        file_response = httpx.get(
-            url_data["download_url"], timeout=30.0, follow_redirects=True
-        )
 
         path = self._file_path(product, item)
```
Comment (glujan) [inline drpg/sync.py:92]: Could you please move this line to the top of this method and rewrite the rest to only have a single one check for running in dry mode? Something like:

```python
path = ...
if self._dry_run:
    ...
else:
    ...
```

**Review by glujan — CHANGES_REQUESTED** [review]
Hi @ToddBradley thanks a lot for this PR, having a dry-run mode makes a lot of sense.

I'm happy to merge this, but I have one small change request before doing that.

File: drpg/sync.py (hunk 1 above)
Comment (ToddBradley) [reply drpg/sync.py]: OK, done. Take a look.

**Review by glujan — APPROVED** [review]
Thanks again 👍

**Pull request merged**